libtransact/src/context/manager/thread/mod.rs

```diff
@@ -0,0 +1,432 @@
+/*
+ * Copyright 2019 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+//! This module provides a threaded ContextManager.
+//!
+
+mod core;
+
+pub use crate::context::manager::thread::core::{
+    spawn_context_manager_thread, ContextManagerJoinHandle, ContextOperationMessage,
+    ContextOperationResponse,
+};
+
+use std::sync::mpsc;
+use std::sync::mpsc::{RecvError, Sender};
+
+use crate::context::error::{ContextManagerCoreError, ContextManagerError};
+use crate::context::manager::thread::core::ContextOperationResult;
+use crate::context::{ContextId, ContextLifecycle};
+use crate::protocol::receipt::{Event, TransactionReceipt};
+
+/// A ContextManager which interacts with a threaded version of the ContextManager operations.
+#[derive(Clone)]
+pub struct ContextManager {
+    /// Used to send ContextOperationMessage to the Context Manager core
+    core_sender: Sender<ContextOperationMessage>,
+}
+
+impl ContextManager {
+    fn new(sender: Sender<ContextOperationMessage>) -> Self {
+        ContextManager {
+            core_sender: sender,
+        }
+    }
+
+    /// Returns a list of key and value pairs from the specified context's state using the
+    ///     list of keys.
```

Comment:
The reverse indent here feels awkward - is this actually a common rustdoc thing?